packages/engine/Source/Shaders/Builtin/Functions/depthClamp.glsl

```diff
@@ -1,6 +1,6 @@
 // emulated noperspective
-#if defined(GL_EXT_frag_depth) && !defined(LOG_DEPTH)
-varying float v_WindowZ;
+#if (__VERSION__ == 300 || defined(GL_EXT_frag_depth)) && !defined(LOG_DEPTH)
+out float v_WindowZ;
 #endif
```

Comment:
Other files will need this treatment including `writeDepthClamp.glsl` and `writeLogDepth.glsl`.

I would do a search in `main` for `ifdef GL_EXT_frag_depth` and `defined(GL_EXT_frag_depth` and replace all those occurrences with `(__VERSION__ == 300 || defined(GL_EXT_frag_depth))`.

E.g. in `main` `Primitive.js` there's

```js
  modifiedFS +=
    "void main() {\n" +
    "    czm_non_depth_clamp_main();\n" +
    "#if defined(GL_EXT_frag_depth)\n" +
    "    #if defined(LOG_DEPTH)\n" +
    "        czm_writeLogDepth();\n" +
    "    #else\n" +
    "        czm_writeDepthClamp();\n" +
    "    #endif\n" +
    "#endif\n" +
    "}\n";
```

and on this branch it's

```js
  modifiedFS +=
    "void main() {\n" +
    "    czm_non_depth_clamp_main();\n" +
    "    #if defined(LOG_DEPTH)\n" +
    "        czm_writeLogDepth();\n" +
    "    #else\n" +
    "        czm_writeDepthClamp();\n" +
    "    #endif\n" +
    "}\n";
```

but really it should be

```js
  modifiedFS +=
    "void main() {\n" +
    "    czm_non_depth_clamp_main();\n" +
    "#if (__VERSION__ == 300 || defined(GL_EXT_frag_depth))\n" +
    "    #if defined(LOG_DEPTH)\n" +
    "        czm_writeLogDepth();\n" +
    "    #else\n" +
    "        czm_writeDepthClamp();\n" +
    "    #endif\n" +
    "#endif\n" +
    "}\n";
```

otherwise I think there's the potential to break the WebGL 1 fallback.

Reply:
Thanks @lilleyse, I added the version/extension check to `writeLogDepth.glsl` and `writeDepthClamp.glsl`, and also `EllipsoidFS.glsl`.
I think these changes make the check in `Primitive.js` unnecessary, since both `czm_writeLogDepth` and `czm_writeDepthClamp` become no-ops if the version/extension check fails.

Reply:
Ah yes, you're right about `Primitive.js`. The latest commit looks good.